d3b_dff_cli/modules/intake/request.py

```diff
@@ -0,0 +1,177 @@
+"""Create data intake epic in Jira"""
+
+import json
+import urllib3
+import logging
+import time
+from datetime import datetime
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def check_status(response):
+    """
+    Check response status and handle appropriately
+    """
+
+    # check that response is successful
+    if response.status in [201, 204]:
+        logger.info("Jira ticket updated successfully")
+    elif response.status == 200:
+        logger.info("Get request successful")
+    else:
+        logger.error(
+            f"Failed to update Jira ticket: {response.status} - {response.data}"
+        )
+        exit(1)
+
+    return
+
+
+def convert_keys_to_id(key, field, meta_res):
+    """
+    Convert text key to value id.
+    Set default values based on "Unknown" or similar value for field.
+    Loop through allowed values to find match and return id.
+
+    Inputs:
+    - key (str): Text key to convert
+    - field (str): Field name
+    - meta_res (urllib3.response.HTTPResponse): Response from Jira API
+
+    Returns:
+    - field_id (str): ID of allowed value
+    """
+
+    # set up defaults
+    default_ids = {
+        "Study": "10412",  # Unknown
+        "Data Source": "10413",  # Source Not Listed - Please Add New Source
+        "Program": "10380",  # D3B
+    }
+
+    field_id = default_ids[field]
+
+    meta_data = json.loads(meta_res.data)
+
+    for ticket_field in meta_data["fields"]:
+        if ticket_field["name"] == field:
+            for option in ticket_field["allowedValues"]:
+                if option["value"] == key:
+                    field_id = option["id"]
+
+    return field_id
+
+
+def intake_request(args):
+    """
+    Create data intake epic and return epic and transfer ticket ids.
+
+    Input:
+    - args (argparse.Namespace): Parsed command-line arguments
+
+    Output:
+    - epic_id (str): Epic ID
+    - transfer_id (str): Transfer ID
+    """
+
+    # jira internal id for the data intake epic
+    intake_epic_id = 10231
+
+    # jira internal id for AD project
+    project_id = 10147
```

Comment:
Same here. Should this be a variable so that we can use this cli tool against other projects. I think it would be good to have this be a variable as well as so tickets can be created across projects. This should then be an input to the cli command that can be set in different jobs.